pyne/mesh.py

```diff
@@ -1510,6 +1509,162 @@ def _do_op(self, other, tags, op, in_place=True):
+    dims : list
+        Dimensions of the mesh.
+    num_ves : int
+        Number of volume elements.
+    e_bounds : list of floats
+        The minimum and maximum bounds for energy bins
+    num_e_groups: int
+        Number of energy groups.
+    mesh :
+        An PyMOAB core instance tagged with all results and
+        relative errors
+    tag_names : iterable
+        Four strs that specify the tag names for the results, relative errors,
+        total results, and relative errors of the total results.
+
+    Notes
+    -----
+    All Mesh/StatMesh attributes are also present via a super() call to
+    StatMesh.__init__().
+
+    """
+
+    def __init__(self):
+        """
+        Create an empty MeshTally object and set default values.
+        """
+
+        if not HAVE_PYMOAB:
+            raise RuntimeError("PyMOAB is not available, "
+                               "unable to create Meshtally Mesh.")
+
+        self.tally_number = None
+        self.particle = 'neutron'
+        self.tag_names = None
+
+
+    def tag_flux_error_from_mcnp_tally_results(self, result, rel_error,
+            res_tot, rel_err_tot):
+        """
+        This function uses the output tally result from mcnp result and
+        rel_error to set the flux and error tags.
+
+        Parameters
+        ----------
+        result : numpy array
+            This numpy array contains the flux data read from MCNP meshtally
+            file. The shape of this numpy array is
+            (num_e_groups*num_ves).
+        rel_error: numpy array
+            This numpy array contains the relative error data read from MCNP
+            meshtally.
+        res_tot : list
+            The total results.
+        rel_err_tot : list
+            Relative error of total results.
+        """
+        # Tag results and error vector to mesh
+        self.tag(self.tag_names[0], tagtype='nat_mesh',
+                 size=self.num_e_groups, dtype=float)
+        res_tag = self.get_tag(self.tag_names[0])
+        self.tag(self.tag_names[1], tagtype='nat_mesh',
+                 size=self.num_e_groups, dtype=float)
+        rel_err_tag = self.get_tag(self.tag_names[1])
+
+        if self.num_e_groups == 1:
+            res_tag[:] = result[0]
+            rel_err_tag[:] = rel_error[0]
+        else:
+            res_tag[:] = result.transpose()
+            rel_err_tag[:] = rel_error.transpose()
+
+        if self.num_e_groups > 1:
+            self.tag(self.tag_names[2], size=1,
+                     dtype=float, tagtype='nat_mesh')
+            res_tot_tag = self.get_tag(self.tag_names[2])
+    
+            self.tag(self.tag_names[3], size=1,
+                     dtype=float, tagtype='nat_mesh')
+            rel_err_tot_tag = self.get_tag(self.tag_names[3])
+    
+            res_tot_tag[:] = res_tot
+            rel_err_tot_tag[:] = rel_err_tot
+
+
+    def tag_flux_error_from_openmc_tally_results(self, result, rel_err,
```

Comment:
It seems like we'll get different tag structures depending on whether it's MCNP or OpenMC. Is that right?  Shouldn't we figure out a single canonical structure?  Maybe that will break backwards compatibility for MCNP?

Reply:
Yes, the results from `MCNP` and `openmc` has different shapes. The one from `MCNP` need to transpose before tagging data to mesh. We can use the same tag method if we make the conversion before tagging.

Reply:
If possible, I think it would be good for mesh tally files to have the same structure regardless of their source

Reply:
Will do.